File: solutions/day16/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findSmallestInteger(vector<int>& nums, int value) {
        int ans = 0;

        vector<int> a(value, 0);

        for (int i = 0; i < nums.size(); i++)
        {
            int x = ((nums[i] % value) + value) % value;
			a[x]++;
            
        }

    
        while(a[ans % value]--){
            ans++;
        } 
	
        

        return ans;
    }
};

File: solutions/day16/solution.py
class Solution:
    def findSmallestInteger(self, nums, value):
        ans=0
        temp=[0]*value

        for num in nums:
            x = ((num % value) + value) % value
            temp[x] += 1

        while temp[ans % value] > 0:
            temp[ans % value] -= 1
            ans+=1

        return ans
